src/main/java/rx/schedulers/Schedulers.java

```diff
@@ -32,7 +35,14 @@
     private final Scheduler ioScheduler;
     private final Scheduler newThreadScheduler;
 
-    private static final Schedulers INSTANCE = new Schedulers();
+    private static Schedulers INSTANCE;
+
+    private static synchronized Schedulers getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Schedulers();
+        }
+        return INSTANCE;
+    }
```

Comment:
so should it just call the static `shutdown()`? Alternatively, the public `shutdown()` could defer to a private instance `shutdown`.

```java
private static Schedulers getInstance() {
    for (;;) {
        Schedulers current = INSTANCE.get();
        if (current != null) {
            return current;
        }
        current = new Schedulers();
        if (INSTANCE.compareAndSet(null, current)) {
            return current;
        } else {
            current.instanceShutdown();
        }
    }
}

public static void shutdown() {
    getInstance().instanceShutdown();
}

private void instanceShutdown() {
    synchronized (this) {
        if (computationScheduler instanceof SchedulerLifecycle) {
            ((SchedulerLifecycle) computationScheduler).shutdown();
        }
        if (ioScheduler instanceof SchedulerLifecycle) {
            ((SchedulerLifecycle) ioScheduler).shutdown();
        }
        if (newThreadScheduler instanceof SchedulerLifecycle) {
            ((SchedulerLifecycle) newThreadScheduler).shutdown();
        }

        GenericScheduledExecutorService.INSTANCE.shutdown();

        RxRingBuffer.SPSC_POOL.shutdown();

        RxRingBuffer.SPMC_POOL.shutdown();
    }
}
```